adam-pipeline/spark-toil-script.py

```diff
@@ -0,0 +1,183 @@
+    job.addFollowOnJobFn(acquire_data, masterIP, outDir, bamNames)
+
+def acquire_data(job, masterIP, dataDir, outDir, bamNames):
+    tachyonNames = []
+    for bamFile in bamNames:
+        fileSystem, path = bamFile.split('://')
+        nameOnly = path.split('/')[-1]
+        call(["docker", "run", "conductor", bamFile, \
+              "hdfs://"+masterIP+":"+TACHYON_MASTER_PORT+"/"+nameOnly])
+        tachyonNames.append(nameOnly)
+    job.addFollowOnJobFn(adam_init, masterIP, outDir, tachyonNames)
+
+def adam_init(job, masterIP, outDir, bamFiles):
+    log.write("adam init\n")
+    log.flush()
+
+    global ADAM_COMMAND
+    ADAM_COMMAND = ["docker", "run", "computationalgenomicslab/adam", "--master", \
+                    "spark://"+masterIP+":"+SPARK_MASTER_PORT, "--", "transform"]
+    global TACHYON_URL
+    TACHYON_URL = "tachyon://"+masterIP+":"+TACHYON_MASTER_PORT
+
+    for bamFile in bamFiles:
+        job.addChildJobFn(bam_to_adam, outDir, bamFile)
+    
+def bam_to_adam(job, outDir, inFile):
+    outFile = '.'.join(inFile.split(".")[:-1])+".adam"
+    call(ADAM_COMMAND + [TACHYON_URL+"/"+, TACHYON_URL+"/"+outFile])
+    job.addChildJobFn(mark_duplicates, outDir, outFile)
+
+def mark_duplicates(job, outDir, inPath, prefix):
+    outFile = '.'.join(inFile.split(".")[:-1])+".markdup.adam"
+    call(ADAM_COMMAND + [TACHYON_URL+"/"+inFile, TACHYON_URL+"/"+outFile, "-mark_duplicate_reads"])
+    job.addChildJobFn(realign_indels, outDir, outFile)
+
+def realign_indels(job, outDir, inFile):
+    outFile = '.'.join(inFile.split(".")[:-2])+".ri.adam"
+    call(ADAM_COMMAND + [TACHYON_URL+"/"+inFile, TACHYON_URL+"/"+outFile, "-realign_indels"])
+    job.addChildJobFn(base_quality_score_recalibration, outDir, outFile)
+
+def base_quality_score_recalibration(job, outDir, inFile):
+    outFile = '.'.join(inFile.split(".")[:-2])+".bqsr.adam"
+    call(ADAM_COMMAND +[TACHYON_URL+"/"+inFile, TACHYON_URL+"/"+outFile, "-recalibrate_base_qualities"])
+    job.addChildJobFn(adam_to_bam, outDir, outFile)
+
+def adam_to_bam(job, outDir, inFile):
+    outFile = '.'.join(inFile.split(".")[:-1])+".bam"
+    call(ADAM_COMMAND +[TACHYON_URL+"/"+inFile, TACHYON_URL+"/"+outFile, "-recalibrate_base_qualities"])
+    job.addFollowOnJobFn(write_data, outDir, outFile)
+
+def write_data(job, outDir, origFile):
+    call(["docker", "run", "conductor", "-a", "AWS_ACCESS_KEY="+accessKey, "-a", "AWS_SECRET_KEY="+secretKey, \
+          "hdfs://"+masterIP+":"+TACHYON_MASTER_PORT+"/"+origFile, outDir+"/"+origFile])
+
+    
+# SERVICE CLASSES
+
+class MasterService(Job.Service):
+
+    def start(self):
+        log.write("start master\n")
+        log.flush()
+        self.sparkContainerID = check_output(["docker", "run", "--net='host'", "-d", "-e",\
+                                              "\"SPARK_MASTER_IP=$(hostname -f)\"", \
+                                              "computationalgenomicslab/apache-spark-master:1.5.0"])[:-1]
```

Comment:
This could be bumped to version 1.5.2, see https://github.com/BD2KGenomics/cgl-docker-lib/pull/58